vite-project/src/utils/api.js

```diff
@@ -95,4 +95,16 @@ export const deleteComment = (numericalId) => {
         err.response ? err.response.data : err.message
       );
     });
-};
\ No newline at end of file
+};
+
+export const getArticlesByTopics = (topic) => {
+  return ncnewsAPI
+   .get(`/articles?topic=${topic}`)
+   .then((res) => {
+      return res.data.allArticles;
+    })
+   .catch((err) => {
+    "Error posting like:",
+        err.response ? err.response.data : err.message
+    });

```

Comment:
This shouldn't need to be done using a completely new function. I would recommend looking into axios params and how this could be used with the original function to get all articles to make it more reusable.